core/src/main/java/nu/yona/server/analysis/entities/IntervalActivity.java

```diff
@@ -98,7 +99,7 @@ public UserAnonymized getUserAnonymized()
 
 	public Goal getGoal()
 	{
-		return goal;
+		return EntityUtil.enforceLoading(goal);
```

Comment:
It's not the caching but the JPA lazy loading.

Honestly, I don't know why I was facing the proxy issue here. The goal is not lazily loaded, so I was expecting it to be available along with the activity, but it wasn't always.